numpy/core/tests/test_regression.py

```diff
@@ -2333,6 +2333,18 @@ def test_void_item_memview(self):
         #del va
         #assert_equal(x, b'\x00\x00\x00\x00')
 
+    def test_reduce_warns(self):
+        #gh 11029
+        # make sure a runtime warning is raised no matter where NAN is
+        with suppress_warnings() as sup:
+            n = 16  # 4 * len(SSE packing)
+            sup.record(RuntimeWarning)
+            for i in range(n):
```

Comment:
You could use the `diagflat` trick here, maybe with different values of n.

Reply:
Maybe let `n` range through various powers of two, say 8, 16, 32.

Reply:
the test is meant to check that a warning is raised for any single NAN no matter where in a vector, the `daigflat` call would only emit one warning, not `n` warnings, no?

fixed to let `n` range through powers of 2

Reply:
You can run it with an iterator
```
for row in diagflat(...):
    whatever
```